test/hotspot/jtreg/runtime/verifier/CFLH/TestVerify.java

```diff
@@ -0,0 +1,183 @@
+                } catch (Throwable e) {
+                    System.out.println(e);
+                    throw new AssertionError(e);
+                }
+                return bytes;
+            }
+            return null;
+        }
+    }
+
+    static Instrumentation inst = null;
+    static boolean verifyErrorThrown = false;
+
+    public static void premain(String args, Instrumentation instrumentation) throws Exception {
+        System.out.println("Premain");
+        inst = instrumentation;
+
+        // double check our class hasn't been loaded yet
+        for (Class clazz : inst.getAllLoadedClasses()) {
+            if (clazz.getName().equals(CLASS_TO_BREAK)) {
+                throw new AssertionError("Oops! Class " + CLASS_TO_BREAK + " is already loaded, the test can't work");
+            }
+        }
+
+        boolean retransform = Boolean.getBoolean("agent.retransform");
+        if (retransform) {
+            // for the bug the class we call must be already loaded, so that we retransform it
+            var clazz = Class.forName(CLASS_TO_BREAK);
+            try {
+                 inst.addTransformer(new BadTransformer(), true);
+                 inst.retransformClasses(clazz);
+             } catch (VerifyError ve) {
+                 System.out.println("Passed: threw VerifyError during retransform");
+                 verifyErrorThrown = true;
+             }
+        } else {
+            // no verify error, the class is used in TestMain
+            inst.addTransformer(new BadTransformer());
+        }
+    }
+
+    private static void buildAgent() {
+        try {
+            ClassFileInstaller.main("TestVerify");
+        } catch (Exception e) {
+            throw new RuntimeException("Could not write agent classfile", e);
+        }
+
+        try {
+            PrintWriter pw = new PrintWriter("MANIFEST.MF");
+            pw.println("Premain-Class: TestVerify");
+            pw.println("Agent-Class: TestVerify");
+            pw.println("Can-Retransform-Classes: true");
+            pw.println("Can-Redefine-Classes: true");
+            pw.close();
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException("Could not write manifest file for the agent", e);
+        }
+
+        sun.tools.jar.Main jarTool = new sun.tools.jar.Main(System.out, System.err, "jar");
+        if (!jarTool.run(new String[] { "-cmf", "MANIFEST.MF", "redefineagent.jar", "TestVerify.class" })) {
+            throw new RuntimeException("Could not write the agent jar file");
+        }
+    }
+
+    public static void main(String argv[]) {
+        if (argv.length == 1 && argv[0].equals("buildAgent")) {
+            buildAgent();
+            return;
+        }
+
+        try {
+            // Now load the class for the VerifyError.
+            System.out.println("1 hour is " + Duration.ofHours(1).getSeconds() + " seconds");
+            if (!verifyErrorThrown) {
```

Comment:
I'm a bit confused about the logic here - are there three possibilities:
1. the load threw VerifyError and will be caught below
2. Another part of the code (where?) threw VerifyError, caught it and set `verifyErrorThrown` to true
3. No verification error occurred at all.

How do we trigger case 1 versus case 2?